staging/kos/pkg/registry/network/subnet/strategy.go

```diff
@@ -154,51 +156,36 @@ func (ss *subnetStrategy) validate(s *network.Subnet) field.ErrorList {
 		return errs
 	}
 
-	// Retrieve the summaries of all the other existing subnets with the same
-	// VNI so that we can check for conflicts.
-	potentialRivals, err := ss.subnetsWithVNI(fmt.Sprint(subnetSummary.VNI))
+	return append(errs, ss.checkNSAndCIDRConflicts(subnetSummary, malformedCIDR)...)
+}
+
+func (ss *subnetStrategy) checkNSAndCIDRConflicts(candidate *subnet.Summary, malformedCIDR bool) (errs field.ErrorList) {
+	potentialRivals, err := ss.subnetIndexer.ByIndex(subnetVNIIndex, strconv.FormatUint(uint64(candidate.VNI), 10))
 	if err != nil {
-		return append(errs, field.InternalError(nil, err))
+		glog.Errorf(".ByIndex failed for index %s and vni %d: %s", subnetVNIIndex, candidate.VNI, err.Error())
```

Comment:
Starting a log message with a dot is awkward.  I suggest starting with "ByIndex" or "Indexer.ByIndex" or "Subnet Indexer.ByIndex" or "subnetIndexer.ByIndex" or "ss.subnetIndexer.ByIndex".

Reply:
Fixed in 63487bb.